src/indexable.cr

```diff
@@ -281,6 +281,17 @@ module Indexable(T)
     self
   end
 
+  # Returns an `Array` with all the elements in the collection.
+  #
+  # ```
+  # {1, 2, 3}.to_a # => [1, 2, 3]
+  # ```
+  def to_a
+    ary = Array(T).new(size)
+    each { |e| ary << e }
```

Comment:
The optimizer should make this into a single memcopy for indexable with the elements in sequence, right?

Reply:
I don't think so. The optimizer is pretty dumb.

Reply:
If this ends up as a bottleneck in any specific class, there's plenty of scope for overriding it to use memcpy in classes where that's possible.